mm2src/coins/utxo/utxo_common.rs

```diff
@@ -1332,31 +1371,52 @@ pub async fn withdraw<T>(coin: T, req: WithdrawRequest) -> WithdrawResult
 where
     T: AsRef<UtxoCoinFields> + UtxoCommonOps + MarketCoinOps,
 {
-    let to = coin
-        .address_from_str(&req.to)
-        .map_to_mm(WithdrawError::InvalidAddress)?;
-
     let conf = &coin.as_ref().conf;
-    let is_p2pkh = to.prefix == conf.pub_addr_prefix && to.t_addr_prefix == conf.pub_t_addr_prefix;
-    let is_p2sh = to.prefix == conf.p2sh_addr_prefix && to.t_addr_prefix == conf.p2sh_t_addr_prefix && conf.segwit;
 
-    let script_pubkey = if is_p2pkh {
-        Builder::build_p2pkh(&to.hash)
-    } else if is_p2sh {
-        Builder::build_p2sh(&to.hash)
+    let script_pubkey = if is_segwit_address(&req.to) {
+        let to = SegwitAddress::from_str(&req.to).unwrap();
+
+        if Some(to.hrp.clone()) != conf.bech32_hrp {
+            let error = format!("Address {} is not a valid bech32 address for {}", to, conf.ticker);
+            MmError::err(WithdrawError::InvalidAddress(error))
+        } else {
+            Ok(Builder::build_p2wpkh(&AddressHash::from(&to.program[..])))
+        }
     } else {
-        let error = "Expected either P2PKH or P2SH".to_owned();
-        return MmError::err(WithdrawError::InvalidAddress(error));
-    };
+        let to = if is_cashaddress(&req.to) {
+            address_from_str_and_format(conf, &req.to, &UtxoAddressFormat::CashAddress {
```

Comment:
This code now allows withdrawing BTC to e.g. cash address format that shouldn't work - e.g. `bchtest:qqgp9xh3435xamv7ghct8emer2s2erzj8gx3gnhwkq` address for tBTC withdrawal. Please add a test for this case and fix it.